### DISCUSSION TITLE
Serializing the schema model for form validation

### DISCUSSION THREAD

**Discussion by avifatal** [rsuite/rsuite#2082] [General]
Hi,
I want to pass the form model from the server.
I dont want to send it like that:
```
const model = Schema.Model({
  email: Schema.Types.StringType().isEmail('Please enter a valid email address.')
});
```

I preffer to pass json... something like:
```
const model = Schema.Model({
  email: {type: 'string', validations:['method' : 'isEmail', message: 'Please enter a valid email address.']}
});
```

Is there some way of doing that?
Thanks

**Comment by SevenOutman** [comment]
See if [rsuite/validation](https://github.com/rsuite/validation) matches what you need.

Or you may try JSON Schema which is a universal format of describing data structure including validation rules (used by OpenAPI Specification and SchemeStore).

**Reply by avifatal** [reply]
@SevenOutman thank you.
This is perfect, but there is a compilation error:
```
./node_modules/@rsuite/validation/dist/adaptors/BaseTypeAdaptor.js
Module not found: Can't resolve 'rsuite/lib/Schema' in '/home/avi/work/skyline/react-app/node_modules/@rsuite/valid
ation/dist/adaptors'
```
The code:
```
import {Validator} from "@rsuite/validation";

    const model = Validator.SchemaModel({
        name: "required",
        email: "required|email",
    });
 <Form layout={"inline"} model={model}>....
```
Thanks

**Reply by SevenOutman** [reply]
Ooops, looks like rsuite/validation is not yet compatible with RSuite v5 :( We need to upgrade it some time.

**Reply by avifatal** [reply]
Is it simple? do you have estimation for that?
I would contribute to that, but my react knowledge is very limited... mostly integrations...
thanks

**Reply by SevenOutman** [reply]
It won't be too sophisticated. I think I'd do that tomorrow.

**Reply by SevenOutman** [reply]
I have published an experimental upgrade. You can try installing it with

     npm install @rsuite/validation@next

**Reply by avifatal** [reply]
Thank you!
